sharktank/sharktank/models/deepseek/toy_deepseek.py

```diff
@@ -22,33 +22,46 @@
 def generate(seed):
     torch.manual_seed(seed=12345)
     dtype = torch.float32
-    block_seq_stride = 16
+    block_seq_stride = 32
     max_blocks = 8
 
-    attn_head_dim = 16
-    rope_dimension_count = 16
+    rope_dimension_count = 64
     vocabulary_size = 256
     expert_count = 4
     used_experts = 2
+    qk_nope_head_dim = 128
+    qk_rope_head_dim = 64
+    attn_head_dim = qk_nope_head_dim + qk_rope_head_dim
 
     config = LlamaModelConfig(
         hp=LlamaHParams(
+            model_arch="deepseek2",
             context_length=block_seq_stride * max_blocks,
             embedding_length=32,
-            block_count=1,
+            block_count=4,
             feed_forward_length=23,
-            rope_dimension_count=rope_dimension_count,
-            rope_freq_base=500000.0,
             attention_head_count=4,
             attn_head_dim=attn_head_dim,
-            attention_layer_norm_rms_epsilon=0.01,
+            attention_layer_norm_rms_epsilon=9.0,
             attention_head_count_kv=4,
-            nope_dim=32,
-            kv_latent_dim=16,
-            v_head_dim=32,
+            q_lora_rank=1536,
+            kv_lora_rank=512,
+            qk_nope_head_dim=qk_nope_head_dim,
+            qk_rope_head_dim=qk_rope_head_dim,
+            v_head_dim=128,
+            rope_dimension_count=rope_dimension_count,
+            rope_freq_base=10000.0,
             expert_count=expert_count,
             expert_used_count=used_experts,
-            model_arch="deepseek2",
+            expert_shared_count=1,
+            n_expert_groups=8,
+            n_limited_groups=4,
```

Comment:
```suggestion
            n_expert_groups=2,
            n_limited_groups=2,
```
Current values are invalid, too large.

Reply:
Updated